csrc/ir/printer.cpp

```diff
@@ -84,6 +84,18 @@ void IrPrinter::handle(const hir::HostIrContainer* host_ir_container) {
     os() << host_unit->toString(indent_size_);
   }
   os() << "} // %HostIrContainer\n\n";
+
+  // Print the definitions of the indices that are used in the host_ir_container
+  os() << "Index definitions:\n";
```

Comment:
Instead of printing these unconditionally, each DebugDumpOption can take a list of arguments to tune the verbosity level. For example, I'd like `NVFUSER_DUMP=host_ir` to print out the top-levels and `NVFUSER_DUMP='host_ir(indices)'` to print out index definitions in addition. 

If you are willing to do that in this PR, great! If not, totally fine and I'll find some time to add this myself. 

https://github.com/NVIDIA/Fuser/blob/618673aa1cf929918c97d41ed6ebdb2c8be8f5a3/csrc/options.h#L281

Reply:
ok, I didn't know about that, thanks for pointing out. I made the change as you suggest. Ideally I think there should be an option to `HostIrContainer::print()` but I rather made `print.cpp` read the env var because it is simpler and doesn't change any function signature